x-pack/solutions/security/plugins/elastic_assistant/server/lib/prompt/get_prompt.ts

```diff
@@ -0,0 +1,222 @@
+ * @param actionsClient - actions client
+ * @param connector - connector, provide if available. No need to provide model and provider in this case
+ * @param connectorId - connector id
+ * @param model - model. No need to provide if connector provided
+ * @param promptGroupId - feature id, should be common across promptIds
+ * @param promptIds - prompt ids with shared promptGroupId
+ * @param provider  - provider. No need to provide if connector provided
+ * @param savedObjectsClient - saved objects client
+ */
+export const getPromptsByGroupId = async ({
+  actionsClient,
+  connector,
+  connectorId,
+  model: providedModel,
+  promptGroupId,
+  promptIds,
+  provider: providedProvider,
+  savedObjectsClient,
+}: GetPromptsByGroupIdArgs): Promise<PromptArray> => {
+  const { provider, model } = await resolveProviderAndModel({
+    providedProvider,
+    providedModel,
+    connectorId,
+    actionsClient,
+    providedConnector: connector,
+  });
+
+  const prompts = await savedObjectsClient.find<Prompt>({
+    type: promptSavedObjectType,
+    searchFields: ['promptGroupId'],
+    search: promptGroupId,
+  });
+  const promptsOnly = prompts?.saved_objects.map((p) => p.attributes) ?? [];
+
+  return promptIds.map((promptId) => {
+    const prompt = findPromptEntry({
+      prompts: promptsOnly.filter((p) => p.promptId === promptId) ?? [],
+      promptId,
+      promptGroupId,
+      provider,
+      model,
+    });
+    if (!prompt) {
+      throw new Error(
+        `Prompt not found for promptId: ${promptId} and promptGroupId: ${promptGroupId}`
+      );
+    }
+
+    return {
+      promptId,
+      prompt,
+    };
+  });
+};
+
+/**
+ * Get prompt by promptId
+ * provide either model + provider or connector to avoid additional calls to get connector
+ * @param actionsClient - actions client
+ * @param connector - connector, provide if available. No need to provide model and provider in this case
+ * @param connectorId - connector id
+ * @param model - model. No need to provide if connector provided
+ * @param promptId - prompt id
+ * @param promptGroupId - feature id, should be common across promptIds
+ * @param provider  - provider. No need to provide if connector provided
+ * @param savedObjectsClient - saved objects client
+ */
+export const getPrompt = async ({
+  actionsClient,
+  connector,
+  connectorId,
+  model: providedModel,
+  promptGroupId,
+  promptId,
+  provider: providedProvider,
+  savedObjectsClient,
+}: GetPromptArgs): Promise<string> => {
+  const { provider, model } = await resolveProviderAndModel({
+    providedProvider,
+    providedModel,
+    connectorId,
+    actionsClient,
+    providedConnector: connector,
+  });
+
+  const prompts = await savedObjectsClient.find<Prompt>({
+    type: promptSavedObjectType,
+    filter: `${promptSavedObjectType}.attributes.promptId: ${promptId} AND ${promptSavedObjectType}.attributes.promptGroupId: ${promptGroupId}`,
```

Comment:
Consider wrapping the `filter`'s `promptId` and `promptGroupId` in quotes, for example:

```typescript
filter: `${promptSavedObjectType}.attributes.promptId: "${promptId}" AND ${promptSavedObjectType}.attributes.promptGroupId: "${promptGroupId}"`,
```